Mathlib/CategoryTheory/Abelian/GrothendieckCategory.lean

```diff
@@ -0,0 +1,54 @@
+/-
+Copyright (c) 2024 Paul Reichert. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Paul Reichert
+-/
+
+import Mathlib.CategoryTheory.Abelian.Subobject
+import Mathlib.CategoryTheory.Abelian.GrothendieckAxioms
+import Mathlib.CategoryTheory.Adjunction.AdjointFunctorTheorems
+
+/-!
+
+# Grothendieck categories
+
+This file defines Grothendieck categories and proves basic facts about them.
+
+## Definitions
+
+A `GrothendieckCategory` is an abelian category provided that it has `AB5` and a separator.
+
+## Theorems
+
+Relevant implications of `GrothendieckCategory` are established in `GrothendieckCategory.hasLimits`
+and `GrothendieckCategory.hasColimits`.
+
+## References
+
+* [Stacks: Grothendieck's AB conditions](https://stacks.math.columbia.edu/tag/079A)
+
+-/
+
+namespace CategoryTheory
+
+open Limits
+
+universe u v
+variable (C : Type u) [Category.{v} C]
+
+/--
+An abelian category `C` is called a Grothendieck category provided that it has `AB5` and a
+separator (see `HasSeparator`).
+-/
+class GrothendieckCategory [Abelian C] [HasFilteredColimits C] [AB5 C] [HasSeparator C] : Prop where
```

Comment:
Could you add a `stacks` attribute? https://leanprover-community.github.io/mathlib4_docs/Mathlib/Tactic/StacksAttribute.html

Reply:
Done -- while I was at it, I also added `stacks` attributes to the AB conditions (`GrothendieckAxioms.lean`). More specifically, I referenced [079B](https://stacks.math.columbia.edu/tag/079B), which contains the definitions of the AB axioms as well as Grothendieck categories.